internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/server/VeniceServerGrpcTest.java

```diff
@@ -0,0 +1,87 @@
+package com.linkedin.venice.server;
+
+import com.linkedin.venice.listener.grpc.VeniceReadServiceClient;
+import com.linkedin.venice.listener.grpc.VeniceReadServiceServer;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class VeniceServerGrpcTest {
+  private VeniceReadServiceServer server;
+  private VeniceReadServiceClient client;
+
+  private void createServer(int port) {
+    this.server = new VeniceReadServiceServer(port);
+    this.server.start();
+  }
+
+  private void createClient(String addr, int port) {
+    this.client = new VeniceReadServiceClient(addr, port);
+  }
+
+  private void stopServer() {
+    this.server.stop();
+  }
+
+  private void shutdownClient() {
+    this.client.shutdown();
+  }
+
+  @Test
+  public void testValidToken() {
+    createServer(50051);
+
+    createClient("localhost", 50051);
+    String value = client.get("valid");
+    Assert.assertEquals(value, "validvalid"); // get function simply returns the key twice, there is no accessing data
+                                              // yet.
+
+    stopServer();
+    shutdownClient();
+  }
+
+  @Test
+  public void testServerNotStarted() {
```

Comment:
Consider using `@Test(expectedExceptions = ..) ` instead of `Assert.fail()`

Your test will look something like this:
```java
@Test(expectedExceptions = StatusRuntimeException.class, expectedExceptionsMessageRegExp = "UNAVAILABLE: io exception")
  public void testServerNotStarted() {
    client.get("valid");
  }
  ```